bfvmm/src/serial/tests/test_serial_port_pl011.cpp

```diff
@@ -0,0 +1,234 @@
+//
+// Bareflank Hypervisor
+//
+// Copyright (C) 2015 Assured Information Security, Inc.
+// Author: Rian Quinn        <quinnr@ainfosec.com>
+// Author: Brendan Kerrigan  <kerriganb@ainfosec.com>
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+
+#include <catch/catch.hpp>
+
+#include <serial/serial_port_pl011.h>
+#include <hippomocks.h>
+#include <map>
+
+#ifdef _HIPPOMOCKS__ENABLE_CFUNC_MOCKING_SUPPORT
+
+static std::map<serial_port_pl011::port_type, serial_port_pl011::value_type> g_ports;
+
+static uint32_t
+mock_ind(uint16_t port) noexcept
+{
+    return g_ports[port];
+}
+
+static void
+mock_outd(uint16_t port, uint32_t val) noexcept
+{
+    g_ports[port] = val;
+}
+
+static void
+mock_serial(MockRepository &mocks)
+{
+    mocks.OnCallFunc(_outd).Do(mock_outd);
+    mocks.OnCallFunc(_ind).Do(mock_ind);
+}
+
+TEST_CASE("serial: constructor_null_intrinsics")
+{
+    MockRepository mocks;
+    mock_serial(mocks);
+
+    CHECK_NOTHROW(std::make_unique<serial_port_pl011>());
+}
+
+
```

Comment:
please remove extra spaces. If you could find a way to auto detect this, I'd love to see another CI test for this as it happens a lot (even from me)

Reply:
clang-format can do this with `MaxEmptyLinesToKeep`, we could always ditch astyle for clang-format since we use clang anyway. That'd require translating our astyle configuration though. If you prefer simple and cowboyish, the following awk script will print the number of extra blank lines in a file...

    awk '/^[ \t]*$/ {if (s) n++; s++} /[^ \t]/ {s = 0} END {print n||0}' < filename

Reply:
We have used clang format in the past and it's not as good as astyle, so I would rather not change. Although the awk script is long, it would work fine for Travis. It would be nice if it would tell you which line numbers are the issue.

Reply:
lol, I wasn't expecting you to be so okay with an ugly awk script. Here's a version that prints line numbers of extra lines - would be easy to pipe this through something else to format it nicely:

    awk '/^[ \t]*$/ {if (s) {n++; print NR;} s++} /[^ \t]/ {s = 0} END {exit (n != 0)}' < filename

It also exits nonzero if there are any extra lines so it can be used easily as a condition in a shell script.